fix(build): escape `install_path` and `lib_dir` in `build.zig`

eamonburns opened the pull request:
Problem:

`zig build` fails on Windows due to `b.install_path` and `b.lib_dir` containing backslashes.

Solution:

Escape each path before interpolating into C file template.

eamonburns committed:
fix(build): escape `install_path` and `lib_dir` in `build.zig`

Problem:

`zig build` fails on Windows due to `b.install_path` and `b.lib_dir` containing backslashes.

Solution:

Escape each path before interpolating into C file template.

justinmk commented on build.zig:
```diff
@@ -229,11 +229,17 @@ pub fn build(b: *std.Build) !void {
     _ = gen_config.addCopyFile(sysconfig_step.getOutput(), "auto/config.h"); // run_preprocessor() workaronnd
+
+    // Escape install_path and lib_dir to include in C string literals (on Windows)
+    const escaped_install_path = try std.mem.replaceOwned(u8, b.allocator, b.install_path, "\\", "\\\\");
```
Can we just force the `\` slashes to be `/` slashes?

eamonburns replied:
I think that would break if there were backslashes in a Unix filename.

Actually, I think the current code breaks when there are backslashes inside a filename, and this PR might fix that.

eamonburns committed:
fix(build): forward slashes instead of escaped backslashes

Review comment: <https://github.com/neovim/neovim/pull/37111#discussion_r2656982207>

bfredl commented on build.zig:
```diff
@@ -229,11 +229,17 @@ pub fn build(b: *std.Build) !void {
+
+    // Escape install_path and lib_dir to include in C string literals (on Windows)
+    const forward_slash_install_path = try std.mem.replaceOwned(u8, b.allocator, b.install_path, "\\", "/");
+    defer b.allocator.free(forward_slash_install_path);
```
just use `b.graph.arena` instead of `b.allocator`, then we don't need defer:s.

eamonburns committed:
fix: use existing function to escape backslashes on Windows

Also document the function

justinmk commented on build.zig:
```diff
@@ -229,11 +229,12 @@ pub fn build(b: *std.Build) !void {
         \\char *default_vimruntime_dir = "";
         \\char *default_lib_dir = "{s}/nvim";
-    , .{ b.install_path, b.lib_dir })); // b.lib_dir is typically b.install_path + "/lib" but may be overridden
+    , .{ try replace_backslashes(b, b.install_path), try replace_backslashes(b, b.lib_dir) }));
```
why delete the comment? it can go above this line

eamonburns committed:
re-add comment about b.lib_dir

eamonburns commented:
Re-added the comment and rebased
